fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/exceptionhandlers/FedoraExceptionMapper.java

```diff
@@ -0,0 +1,163 @@
+     */
+    protected FedoraExceptionMapper(final Response.Status status) {
+        this(status, DEFAULT_CONTENT_TYPE);
+    }
+
+    /**
+     * Alternate constructor to be used by subclasses that wish to override the default status and contentType.
+     * 
+     * @param status http status
+     * @param contentType the content type
+     */
+    protected FedoraExceptionMapper(final Response.Status status, final String contentType) {
+        this.status = status;
+        this.contentType = contentType;
+    }
+
+    @Override
+    public Response toResponse(final T e) {
+        logError(e);
+        debugException(this, e, LOGGER);
+        ResponseBuilder builder = status(e);
+        builder = entity(builder, e);
+        builder = contentType(builder, e);
+        builder = links(builder,e);
+        return builder.build();
+    }
+
+    /**
+     * Sets the content on the builder. Subclasses can override this method in order to define custom logic for
+     * resolving the appropriate contentType.
+     * 
+     * @param builder The response builder.
+     * @param e The exception passed into the mapper.
+     * @return response builder
+     */
+    protected ResponseBuilder contentType(final ResponseBuilder builder, final T e) {
+        return builder.type(contentType);
+    }
+
+    /**
+     * Sets the entity on the builder. Subclasses can override this method in order to define custom logic for
+     * generating the appropriate entity.
+     * 
+     * @param builder The response builder.
+     * @param e The exception passed into the mapper.
+     * @return response builder
+     */
+    protected ResponseBuilder entity(final ResponseBuilder builder, final T e) {
+        return builder.entity(e.getMessage());
+    }
+
+    /**
+     * Set any links the entity on the builder. By default no links are set. Subclasses can override this method in
+     * order to define custom logic for generating the appropriate links.
+     * 
+     * @param builder The response builder.
+     * @param e The exception passed into the mapper.
+     * @return response builder
+     */
+    protected ResponseBuilder links(final ResponseBuilder builder, final T e) {
+        return builder;
+    }
+
+    /**
+     * Logs an error with default formatting. Subclasses may override to provide custom error message formatting.
+     * 
+     * @param e The exception passed into the mapper.
+     */
+    protected void logError(final T e) {
+        LOGGER.error("{} caught an exception: {}", getClass().getSimpleName(), e.getMessage());
+    }
+
+    /**
+     * Creates a response builder with the status defined in the constructor (or the default value, ie BAD_REQUEST, if
+     * the default constructor was used). Subclasses may override to specify a custom status.
+     * 
+     * @param e The exception passed into the mapper.
+     * @return The response builder.
+     */
+    protected ResponseBuilder status(final T e) {
```

Comment:
The idea is that subclasses can inspect the T e and change the status accordingly.